proposals/v3/api.md

```diff
@@ -0,0 +1,1265 @@
+| *capabilities* | Specifies the actions which the device is able to perform (e.g. read, write). See below for more. |
+| *output* | A list of the output types that the device supports. |
+
+
+**Capabilities Fields**
+
+| Field | Description |
+| :---- | :---------- |
+| *mode* | A string specifying the device capabilities. This can be "r" (read only), "rw" (read write), "w" (write only). |
+| *read* | Any additional information regarding the device reads. This will currently remain empty. |
+| *write* | Any additional information regarding device writes. |
+| *write.actions* | A list of actions which the device supports for writing. |
+
+
+**Output Fields**
+
+| Field | Description |
+| :---- | :---------- |
+| *name* | The name of the output type. This can be namespaced. |
+| *type* | The type of the output, as defined by the plugin. |
+| *precision* | The number of decimal places the value will be rounded to. |
+| *scaling_factor* | A scaling factor which will be applied to the raw reading value. |
+| *unit* | Information for the reading's unit of measure. |
+| *unit.name* | The complete name of the unit of measure (e.g. "meters per second"). |
+| *unit.symbol* | A symbolic representation of the unit of measure (e.g. m/s). |
+
+##### Error
+See: [Errors](#errors)
+
+* **500** - Catchall processing error
+* **404** - Device not found
+
+
+---
+
+### Read
+```
+GET http://HOST:5000/v3/read
+```
+
+Read data from devices which match the set of provided tags. 
+
+Passing in the `id` tag here is functionally equivalent to using the [read device](#read-device)
+endpoint.
+
+Reading data will be returned for devices which match *all* of the specified tags.
+The contents of prior reads is not necessarily indicative of the content of future
+reads. That is to say, if a plugin terminates and a read command is issued, the
+devices managed by that plugin which would have matched the tags are no longer available
+to Synse (until the plugin comes back up), and as such, can not be read from.
+When the plugin becomes available again, the devices from that plugin are available
+to be read from.
+
+For readability, readings are sorted by a combination of originating plugin, any
+plugin-specified sort index on the reading's device, and by device id.
+
+For more details on the changes to the `/read` endpoint, see the 
+[Synse Reads Document](./reads.md)
+
+#### Query Parameters
+
+| Key  | Description |
+| :--- | :---------- |
+| ns | The default namespace to use for the specified labels. (default: `default`) |
+| tags | The [tags](tags.md) to filter devices on. If specifying multiple tags, they should be comma-separated. |
+
+#### Response
+It is recommended to read the document on [v3 reads](reads.md), as it explains in detail
+the changes to the read response.
+
+##### OK (200)
+> **Note**: In the examples below, the key-value pairs in the `context` field
+> are arbitrary and only serve as an example of possible values.
+
+```json
+[
```

Comment:
Is there any advantage for keeping the same schema for read and read device here? Since read device only return a unique device, can we just return an object instead of a list with one object?

Reply:
yes -- the returns here aren't 1:1 for a device, in this example there are two readings for device `929b923de65a811`. Each element in the response list is a reading, and a single device may have multiple readings, so even if we are reading from a single device, we'll need to return a list (e.g. read from an LED can return color and state)